cmd/rm-main.go

```diff
@@ -121,17 +133,34 @@ func checkRmSyntax(ctx *cli.Context) {
 	isForce := ctx.Bool("force")
 	isRecursive := ctx.Bool("recursive")
 	isStdin := ctx.Bool("stdin")
-
+	isDangerous := ctx.Bool("dangerous")
+	isNamespaceRemoval := false
+	for _, url := range ctx.Args() {
+		url = filepath.Clean(url)
```

Comment:
you could use `path.Clean()` to work with MS Windows.

Reply:
doesn't filepath.Clean do the same?

Reply:
no

Reply:
path.Clean() doesn't do anything on windows @balamurugana @poornas - filepath.Clean() should be used for cross platform friendly behavior.

Reply:
See below.  This had been discussed long back

```go
package main

import (
	"fmt"
	"path"
	"path/filepath"
)

func main() {
	fmt.Println("filepath.Clean():")
	fmt.Println(filepath.Clean("/s3"))
	fmt.Println(filepath.Clean("s3/"))
	fmt.Println(filepath.Clean("s3/mybucket//object//"))
	fmt.Println(filepath.Clean(`\s3`))
	fmt.Println(filepath.Clean(`s3\`))
	fmt.Println(filepath.Clean(`s3\mybucket\\object\\`))

	fmt.Println("path.Clean():")
	fmt.Println(path.Clean("/s3"))
	fmt.Println(path.Clean("s3/"))
	fmt.Println(path.Clean("s3/mybucket//object//"))
	fmt.Println(path.Clean(`\s3`))
	fmt.Println(path.Clean(`s3\`))
	fmt.Println(path.Clean(`s3\mybucket\\object\\`))
}
```

On MS Windows
```
filepath.Clean():
\s3
s3
s3\mybucket\object
\s3
s3
s3\mybucket\object
path.Clean():
/s3
s3
s3/mybucket/object
\s3
s3\
s3\mybucket\\object\\
```

On GNU/Linux

```
filepath.Clean():
/s3
s3
s3/mybucket/object
\s3
s3\
s3\mybucket\\object\\
path.Clean():
/s3
s3
s3/mybucket/object
\s3
s3\
s3\mybucket\\object\\
```

Reply:
i don't think so you understand the problem of using path.Clean() , path.Clean() doesn't work on windows since it never cleans paths such as `\\` or `C:\mypath\...\otherdir\` . path.Clean() doesn't understand C:\ etc. so it would be wrong to use path.Clean() for cross platform reasons. 

For `mc` we do support such paths and need to behave properly for windows natively once we do filepath.Clean() if we want all the `\` to `/` then we should use filepath.ToSlash() to convert the paths properly.

Using code like in this manner `filepath.ToSlash(filepath.Clean(p))` would work for all operating systems.

Reply:
How is it possible to store object name containing `\`?

Reply:
filepath.ToSlash is used to do proper Windows translation, filepath.ToSlash has no effect on Unixes. Now if you are arguing that if there are files which have `\` on Linux then they are stored fine with this mechanism as well.  Now on windows there is no filename which explicitly contains `\` since its a separator. 

Here is how the behavior is 

- filepath.ToSlash converts all `\` on Windows to `/` 
- filepath.ToSlash converts all `/` to '/` on Unixes. 

filepath.ToSlash is not ubiquitously converting `\` to `/` - https://play.golang.org/p/fFPdEZoAEf